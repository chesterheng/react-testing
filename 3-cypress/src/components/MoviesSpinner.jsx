import React from 'react';
import { Spinner } from 'react-bootstrap';

const MoviesSpinner = () => {
  return (
    <Spinner animation="border" role="alert">
      <span className="sr-only">Content is loading...</span>
    </Spinner>
  )
}

export default MoviesSpinner;
