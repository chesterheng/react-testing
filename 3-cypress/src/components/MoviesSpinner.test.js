import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import MoviesSpinner from "./MoviesSpinner";

describe("MoviesSpinner component", () => {
  test("should render and test MoviesSpinner component", async () => {
    // 2. Render a component
    render(<MoviesSpinner />);

    // 3. Debug DOM state
    // screen.debug();

    // 4. Test spinner component
    const spinner = screen.getByRole("alert");
    // screen.debug(spinner);
    expect(spinner).toHaveTextContent("Content is loading...");
  });

  test("should render and test MoviesSpinner component snapshot", async () => {
    // 2. Render a component
    const { container } = render(<MoviesSpinner />);

    // 3. Debug DOM state
    // screen.debug();

    // 5. Create a snapshot test
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toMatchInlineSnapshot(`
      <div
        class="spinner-border"
        role="alert"
      >
        <span
          class="sr-only"
        >
          Content is loading...
        </span>
      </div>
    `);
  });
});
