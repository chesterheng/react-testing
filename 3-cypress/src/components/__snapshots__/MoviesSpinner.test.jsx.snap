// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesSpinner component should render and test MoviesSpinner component snapshot 1`] = `
<div
  class="spinner-border"
  role="alert"
>
  <span
    class="sr-only"
  >
    Content is loading...
  </span>
</div>
`;
